### PR TITLE
Reland "Android Q transition by default (#82670)"

### LINKED ISSUE #43277: Make Android 10 Activity Animation the default animation
After https://github.com/flutter/flutter/pull/41935 lands, we should default material page transitions to the newer animation.

### LINKED ISSUE #88393: "flutter error messages include a DevTools link" tool test fails
Started failing on https://github.com/flutter/flutter/pull/82670.
```
05:21 +33 ~20 -1: test/integration.shard/overall_experience_test.dart: flutter error messages include a DevTools link [E]                                                                              
  Expected: [
              <a string starting with 'Performing hot reload...'>,
              '',
              '══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════',
              'The following assertion was thrown during layout:',
              'A RenderFlex overflowed by 69200 pixels on the right.',
              '',
              'The relevant error-causing widget was:',
              <match '^  Row .+flutter/dev/integration_tests/ui/lib/overflow\.dart:31:12$'>,
              '',
              'To inspect this widget in Flutter DevTools, visit:',
              <a string starting with 'http'>,
              '',
              'The overflowing RenderFlex has an orientation of Axis.horizontal.',
              'The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and',
              'black striped pattern. This is usually caused by the contents being too big for the RenderFlex.',
              'Consider applying a flex factor (e.g. using an Expanded widget) to force the children of the',
              'RenderFlex to fit within the available space instead of being sized to their natural size.',
              'This is considered an error condition because it indicates that there is content that cannot be',
              'seen. If the content is legitimately bigger than the available space, consider clipping it with a',
              'ClipRect widget before putting it in the flex, or using a scrollable container rather than a Flex,',
              'like a ListView.',
              <match '^The specific RenderFlex in question is: RenderFlex#..... OVERFLOWING:$'>,
              <a string starting with '  creator: Row ← Test ← '>,
              <contains ' ← '>,
              ...
            ]
    Actual: [
              'Performing hot reload...                                        ',
              '',
              '══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════',
              'The following assertion was thrown during layout:',
              'A RenderFlex overflowed by 69200 pixels on the right.',
              '',
              'The relevant error-causing widget was:',
              '  Row Row:file:///b/s/w/ir/x/w/flutter/dev/integration_tests/ui/lib/overflow.dart:31:12',
              '',
              'To inspect this widget in Flutter DevTools, visit:',
              'http://127.0.0.1:9100/#/inspector?uri=http%3A%2F%2F127.0.0.1%3A36561%2FLIw7uQD2nrk%3D%2F&inspectorRef=inspector-0',
              '',
              'The overflowing RenderFlex has an orientation of Axis.horizontal.',
              'The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and',
              'black striped pattern. This is usually caused by the contents being too big for the RenderFlex.',
              'Consider applying a flex factor (e.g. using an Expanded widget) to force the children of the',
              'RenderFlex to fit within the available space instead of being sized to their natural size.',
              'This is considered an error condition because it indicates that there is content that cannot be',
              'seen. If the content is legitimately bigger than the available space, consider clipping it with a',
              'ClipRect widget before putting it in the flex, or using a scrollable container rather than a Flex,',
              'like a ListView.',
              'The specific RenderFlex in question is: RenderFlex#4d88d OVERFLOWING:',
              '  creator: Row ← Test ← Semantics ← Builder ← RepaintBoundary-[GlobalKey#33d8e] ← IgnorePointer ←',
              '    AnimatedBuilder ← Transform ← ScaleTransition ← FadeTransition ← _ZoomExitTransition ← Transform ←',
              ...
            ]
     Which: at location [24] is '    ⋯' which does not match a string ending with ' ← ⋯'
  
  package:test_api                                           expect
  test/integration.shard/overall_experience_test.dart 511:7  main.<fn>
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8838609685218124768/+/u/run_test.dart_for_tool_integration_tests_shard_and_subshard_4_4/test_stdout

Tree is closed.

### PR DESCRIPTION
**Pull request by AlexV525**
Reland #82670 , revert #88394 , fix #88393 and #43277 . /cc @HansMuller @jmagman 

Still waiting for website's instructure to be stable until submit a breaking change note.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/integration.shard/overall_experience_test.dart [hunk 1]
```diff
@@ -533,7 +533,7 @@ void main() {
         matches(RegExp(r'^The specific RenderFlex in question is: RenderFlex#..... OVERFLOWING:$')),
         startsWith('  creator: Row ← Test ← '),
         contains(' ← '),
-        endsWith(' ← ⋯'),
+        endsWith('    ⋯'),
```
Comment (AlexV525) [inline packages/flutter_tools/test/integration.shard/overall_experience_test.dart:536]: This is the fix, caused by the layout changed with the transition.

**Comment by AlexV525** [comment]
Let me know if I should run extra tests during the pre-submit.

**Review by HansMuller — APPROVED** [review]
The original PR - #82670 - caused a tools test failure, which I believe this PR fixes. It also caused a couple of internal golden image test failures which we can safely reset.

**Comment by HansMuller** [comment]
Unfortunately this caused some unexpected golden image failures on internal tests, i.e. not just failures that were obviously due to the route change animation. We're going to have to revert again.

**Pull request merged**